scripts/check-yaml.py

```diff
@@ -49,8 +51,41 @@ def check(self) -> int:
                         "The \"%s\" file must be non-empty",
                         taxonomy.path.with_name(attribution_path.name),
                     )
+                # NOTE: The following three warnings are intended for the
+                # beta only, at the moment, and only to flag issues for
+                # maintainers to address rather than block on them. We will
+                # revisit when other content is allowed.
+                qna_file_path = taxonomy.rel_path.with_name("qna.yaml")
+                if "knowledge" in qna_file_path.parts:
+                    qna_file_contents = parser.parse(qna_file_path).contents
+                    for element in qna_file_contents["document"]["patterns"]:
+                        if not re.match('.*.md', element):
```

Comment:
I thought pdf support was added? Anyway the regex here would be 

`re.search("\.md$", element)`

But what about the pattern `folder_of_md_files/*` which is a legitimate value which should not be rejected.

If you want to do more checking here, I don't think you can do it by pattern matching the yaml contents. You would need to clone the repo, find all files in the repo which match the patterns, and then check that all those files match the desired file types.

Reply:
Yes, pdf support was added in the recent release. Oversight on my part as a newbie.

Technically, the `.*` should match directory patterns when used with `re.match()` and has in the testing I've done, but you're right on the higher level that I probably shouldn't be matching file patterns (I really should be just parsing the string instead of getting into regex at all...) and should be validating the file type from the repo.  Was there a reason the prior version at https://github.com/instructlab/taxonomy/pull/1192 was dropped, or did it just lack contributor time?